### ISSUE TITLE
Internet Usage Monitor

### ISSUE THREAD

**Issue by rayo-alcantar** [nvaccess/addon-datastore#3266]
### Download URL

https://github.com/rayo-alcantar/InternetUsageMonitor/releases/download/2.1.1/InternetUsageMonitor-2.1.1.nvda-addon

### Source URL

https://github.com/rayo-alcantar/InternetUsageMonitor

### Publisher

Angel Alcántar

### Channel

stable

### License Name

GPL v2

### License URL

https://www.gnu.org/licenses/gpl-2.0.html

**Comment by seanbudd** [comment]
@nvdaes can you please add a link to the docs to this message. I think it would be good to provide steps on how to view the sarif

**Comment by seanbudd** [comment]
Also can the PRs with these failures be kept open

**Comment by XLTechie** [comment]
@rayo-alcantar The actual report of what failed, is here:
https://github.com/nvaccess/addon-datastore/actions/runs/8724014019/artifacts/1422432093

@nvdaes, @seanbudd in addition to the docs link in these comments, can we provide the direct link to the zip file as I did above? It is available in the job log after the artifact upload step, so should be returnable.
It would be much more useful than providing the failed job link.

**Comment by seanbudd** [comment]
@rayo-alcantar - this failure is due to the usage of deprecated `mktemp`

You can see how we solved this same problem in NVDA - https://github.com/nvaccess/nvda/pull/16260
In NVDA this was not an actual security issue, and considering this is in a tests folder, we don't think it's a security issue here either.
However, we encourage fixing your add-on to remove the deprecated function in the future.

**Comment by seanbudd** [comment]
@nvdaes - it appears we cannot retrigger this submission as it will always be blocked by the failure.
I think we need to create the PR submission so that NV Access can manually merge when security analysis fails

**Comment by seanbudd** [comment]
@rayo-alcantar - there is also another issue picked up, also in tests, where you use the socket `0.0.0.0` whereas a `127.X.X.X` socket would be more appropriate

**Comment by nvdaes** [comment]
Luke wrote:

> in addition to the docs link in these comments, can we provide the direct link to the zip file as I did above?

The artifact URL is provided in an output of the upload artifact action (artifact-url). The difficulty is that we are analizing, for now,two languages using a matrix (one job for Python and another one for JavaScript). Then I think we may need to create separate comments and the current approach is more simple.
I'm creating a PR with suggested changes, but for now I'mnot addressing this for the mentioned reason.

**Comment by nvdaes** [comment]
BTW, I'll use the provided download URL sent to this issue for testing my next PR, so that the analysis is done before merging the created PR of submitted add-ons.

**Comment by seanbudd** [comment]
This has now been released, however manual approval will need to occur for this add-on until these issues are resolved
